tests/tests/integration_tests.rs

```diff
@@ -108,6 +133,47 @@ async fn ws_unsubscription_works() {
 	assert!(success);
 }
 
+#[tokio::test]
+async fn ws_unsubscription_works_over_proxy_stream() {
+	init_logger();
+
+	let server_addr = server_with_subscription().await;
+	let server_url = format!("ws://{}", server_addr);
+
+	let socks_stream = connect_over_socks_stream(server_addr).await;
+	let data_stream = DataStream::Socks5(socks_stream);
+
+	let client = WsClientBuilder::default()
+		.max_concurrent_requests(1)
+		.build_with_stream(&server_url, data_stream)
+		.await
+		.unwrap();
+
+	let mut sub: Subscription<usize> =
+		client.subscribe("subscribe_foo", rpc_params![], "unsubscribe_foo").await.unwrap();
+
+	// It's technically possible to have race-conditions between the notifications and the unsubscribe message.
+	// So let's wait for the first notification and then unsubscribe.
+	let _item = sub.next().await.unwrap().unwrap();
+
+	sub.unsubscribe().await.unwrap();
+
+	let mut success = false;
+
+	// Wait until a slot is available, as only one concurrent call is allowed.
+	// Then when this finishes we know that unsubscribe call has been finished.
+	for _ in 0..30 {
+		let res: Result<String, _> = client.request("say_hello", rpc_params![]).await;
+		if res.is_ok() {
+			success = true;
+			break;
+		}
+		tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+	}
```

Comment:
So this is all just here to check that unsubscribe has finished; I wonder whether there is a simpler/cleaner way to do this (@niklasad1)

Reply:
yeah but I realized that the subscription doesn't keep the `request id guard` so this doesn't work as intended.

let's remove extra ugliness for the test...